### PR TITLE
fix: Wrong usage of dataloader in GQL group resolver

### PR DESCRIPTION
**Pull request by fregataa**
**Checklist:** (if applicable)

- [x] Milestone metadata specifying the target backport version
- [ ] API server-client counterparts (e.g., manager API -> client SDK)

----
📚 Documentation preview 📚: https://sorna--2056.org.readthedocs.build/en/2056/

----
📚 Documentation preview 📚: https://sorna-ko--2056.org.readthedocs.build/ko/2056/

### CODE REVIEW

**Review by agatha197 — APPROVED** [review]
LGTM

File: src/ai/backend/manager/models/gql.py [hunk 1]
```diff
@@ -913,7 +913,9 @@ async def resolve_group(
                 ctx,
                 "Group.by_user",
             )
-            client_groups = await loader.load(client_user_id, type=[ProjectType[t] for t in type])
+            client_groups = [
+                group for group in await loader.load(client_user_id) if group.type in type
```
Comment (jopemachine) [inline src/ai/backend/manager/models/gql.py:917]: On `groups` table definition, `type` column is non-nullable. May I ask why `null` check on `group.type` is necessary here?

**Comment by jopemachine** [comment]
LGTM, but we also need to edit `resolve_groups`.
https://github.com/lablup/backend.ai/blob/056510689bdec33303dd85d1d0323f5777d5bfdf/src/ai/backend/manager/models/gql.py#L1005

File: src/ai/backend/manager/models/group.py [hunk 2]
```diff
@@ -378,7 +378,8 @@ async def batch_load_by_user(
         cls,
         graph_ctx: GraphQueryContext,
         user_ids: Sequence[uuid.UUID],
-        type: list[ProjectType] = [ProjectType.GENERAL],
+        *,
+        type: list[ProjectType] | None = None,
```
Comment (kyujin-cho) [inline src/ai/backend/manager/models/group.py:382]: Is there a specific reason you decided to expose every types of project by default here? When forming the model store sprint (which is directly related to this `type` parameter) we decided to reveal such projects only if the type was explicitly specified by requester.

**Review by kyujin-cho — CHANGES_REQUESTED** [review]

File: src/ai/backend/manager/models/group.py (hunk 2 above)
Comment (fregataa) [reply src/ai/backend/manager/models/group.py:382]: I forgot to add this. it is restored now

**Pull request merged**